### ISSUE TITLE
In v4.2.5 `@scope/` packages are no longer show up

### ISSUE THREAD

**Issue by eamodio** [RSeidelsohn/license-checker-rseidelsohn#78]
I'm not sure when this started, but v.4.2.0 works with `@scope/` packages, but v4.2.5 no longer does.

Here are some of the packages that used to show up and no longer do:
```
@microsoft/fast-element
@microsoft/fast-react-wrapper
@octokit/core
@opentelemetry/api
@opentelemetry/exporter-trace-otlp-http 
@opentelemetry/sdk-trace-base
@vscode/codicons
@vscode/webview-ui-toolkit
```

**Comment by eamodio** [comment]
In further testing, it looks like it was introduced in v4.2.1

**Comment by eamodio** [comment]
It looks like it was caused by the bump in `read-installed-packages` -- it seems to not return a full object for `@scope/` packages and instead just returns a version number.

**Comment by RSeidelsohn** [comment]
Hi @eamodio, this looks like a major bug and I will take care of it today. Thanks a lot for this issue!

**Comment by jakwarrior** [comment]
Hello, I have the same problem here. This is my package.json :

```
 "dependencies": {
    "@angular/animations": "14.3.0",
    "@angular/cdk": "14.2.7",
    "@angular/common": "14.3.0",
    "@angular/compiler": "14.3.0",
    "@angular/core": "14.3.0",
    "@angular/forms": "14.3.0",
    "@angular/material": "14.2.7",
    "@angular/material-moment-adapter": "14.2.7",
    "@angular/platform-browser": "14.3.0",
    "@angular/platform-browser-dynamic": "14.3.0",
    "@angular/router": "14.3.0",
    "geographiclib": "1.52.2",
    "ngx-cookie-service": "14.0.1",
    "ngx-toastr": "16.1.1",
    "ol": "7.3.0",
    "ol-contextmenu": "5.2.1",
    "ol-ext": "4.0.7",
    "rxjs": "7.8.0",
    "zone.js": "0.12.0"
  },
```

I use this command:
`npx license-checker-rseidelsohn --direct --production --markdown --out licenses.md`

My OS is Windows 10 with Node 18.16.0 and npm 9.5.1

With v4.2.0:
```
- [@angular/animations@14.3.0](https://github.com/angular/angular) - MIT
- [@angular/cdk@14.2.7](https://github.com/angular/components) - MIT
- [@angular/common@14.3.0](https://github.com/angular/angular) - MIT
- [@angular/compiler@14.3.0](https://github.com/angular/angular) - MIT
- [@angular/core@14.3.0](https://github.com/angular/angular) - MIT
- [@angular/forms@14.3.0](https://github.com/angular/angular) - MIT
- [@angular/material-moment-adapter@14.2.7](https://github.com/angular/components) - MIT
- [@angular/material@14.2.7](https://github.com/angular/components) - MIT
- [@angular/platform-browser-dynamic@14.3.0](https://github.com/angular/angular) - MIT
- [@angular/platform-browser@14.3.0](https://github.com/angular/angular) - MIT
- [@angular/router@14.3.0](https://github.com/angular/angular) - MIT
- [@mbda/titan-web-lib@14.0.1](undefined) - Custom: https://nx.dev
- [myApp@1.0.0](undefined) - UNLICENSED
- [geographiclib@1.52.2](https://github.com/geographiclib/geographiclib-js) - MIT
- [ngx-cookie-service@14.0.1](https://github.com/stevermeister/ngx-cookie-service) - MIT
- [ngx-toastr@16.1.1](https://github.com/scttcper/ngx-toastr) - MIT
- [ol-contextmenu@5.2.1](https://github.com/jonataswalker/ol-contextmenu) - MIT
- [ol-ext@4.0.7](https://github.com/Viglino/ol-ext) - BSD-3-Clause
- [ol@7.3.0](https://github.com/openlayers/openlayers) - BSD-2-Clause
- [rxjs@7.8.0](https://github.com/reactivex/rxjs) - Apache-2.0
- [zone.js@0.12.0](https://github.com/angular/angular) - MIT
```

With v4.2.5:
```
- [myApp@1.0.0](undefined) - UNLICENSED
- [geographiclib@1.52.2](https://github.com/geographiclib/geographiclib-js) - MIT
- [ngx-cookie-service@14.0.1](https://github.com/stevermeister/ngx-cookie-service) - MIT
- [ngx-toastr@16.1.1](https://github.com/scttcper/ngx-toastr) - MIT
- [ol-contextmenu@5.2.1](https://github.com/jonataswalker/ol-contextmenu) - MIT
- [ol-ext@4.0.7](https://github.com/Viglino/ol-ext) - BSD-3-Clause
- [ol@7.3.0](https://github.com/openlayers/openlayers) - BSD-2-Clause
- [rxjs@7.8.0](https://github.com/reactivex/rxjs) - Apache-2.0
- [zone.js@0.12.0](https://github.com/angular/angular) - MIT
```

Same problems when I remove _--production_ and _--direct_ options.

**Comment by eamodio** [comment]
@RSeidelsohn Sorry for the delay -- you can see the issue in action on the GitLens repository, I've pushed a [`feature/license-checker`](https://github.com/gitkraken/vscode-gitlens/tree/feature/license-checker) branch with the version upgraded to 4.25, and the resulting incorrect [`ThirdPartyNotices.txt`](https://github.com/gitkraken/vscode-gitlens/blob/e016ce081306dd9bfcdd5faf127e82d11773c0ec/ThirdPartyNotices.txt) vs the correct version on [`main` ](https://github.com/gitkraken/vscode-gitlens/blob/main/ThirdPartyNotices.txt)

To repo:
 - Clone the [repo](https://github.com/gitkraken/vscode-gitlens)
 - run `yarn`
 - run `yarn run update-licenses`

The `update-licenses` script calls into the license checker here:

https://github.com/gitkraken/vscode-gitlens/blob/304dcb2f087c85dd454d0a88f9c9c53775bbf631/scripts/generateLicenses.mjs#L86-L105

Hope that helps!

**Comment by RSeidelsohn** [comment]
Ok, @jakwarrior, I tested your setup on my Ubuntu `22.04.2-LTS` with Node `v18.13.0` nad npm `8.19.3` with this command:
`npx license-checker-rseidelsohn --start ./test --markdown --out licenses.md` - the `./test` directory is where I put the `package.json` with the exact same dependencies as you have, and got this result:

```bash
- [@angular/animations@14.3.0](https://github.com/angular/angular) - MIT
- [@angular/cdk@14.2.7](https://github.com/angular/components) - MIT
- [@angular/common@14.3.0](https://github.com/angular/angular) - MIT
- [@angular/compiler@14.3.0](https://github.com/angular/angular) - MIT
- [@angular/core@14.3.0](https://github.com/angular/angular) - MIT
- [@angular/forms@14.3.0](https://github.com/angular/angular) - MIT
- [@angular/material-moment-adapter@14.2.7](https://github.com/angular/components) - MIT
- [@angular/material@14.2.7](https://github.com/angular/components) - MIT
- [@angular/platform-browser-dynamic@14.3.0](https://github.com/angular/angular) - MIT
- [@angular/platform-browser@14.3.0](https://github.com/angular/angular) - MIT
- [@angular/router@14.3.0](https://github.com/angular/angular) - MIT
- [@mapbox/jsonlint-lines-primitives@2.0.2](https://github.com/mapbox/jsonlint) - UNKNOWN
- [@mapbox/mapbox-gl-style-spec@13.28.0](https://github.com/mapbox/mapbox-gl-js) - ISC
- [@mapbox/point-geometry@0.1.0](https://github.com/mapbox/point-geometry) - ISC
- [@mapbox/unitbezier@0.0.0](https://github.com/mapbox/unitbezier) - BSD-2-Clause
- [@petamoriken/float16@3.8.0](https://github.com/petamoriken/float16) - MIT
- [csscolorparser@1.0.3](https://github.com/deanm/css-color-parser-js) - MIT
- [earcut@2.2.4](https://github.com/mapbox/earcut) - ISC
- [geographiclib@1.52.2](https://github.com/geographiclib/geographiclib-js) - MIT
- [geotiff@2.0.7](https://github.com/geotiffjs/geotiff.js) - MIT
- [ieee754@1.2.1](https://github.com/feross/ieee754) - BSD-3-Clause
- [json-stringify-pretty-compact@2.0.0](https://github.com/lydell/json-stringify-pretty-compact) - MIT
- [lerc@3.0.0](https://github.com/Esri/lerc) - Apache-2.0
- [mapbox-to-css-font@2.4.2](https://github.com/openlayers/mapbox-to-css-font) - BSD-2-Clause
- [minimist@1.2.8](https://github.com/minimistjs/minimist) - MIT
- [moment@2.29.4](https://github.com/moment/moment) - MIT
- [ngx-cookie-service@14.0.1](https://github.com/stevermeister/ngx-cookie-service) - MIT
- [ngx-toastr@16.1.1](https://github.com/scttcper/ngx-toastr) - MIT
- [ol-contextmenu@5.2.1](https://github.com/jonataswalker/ol-contextmenu) - MIT
- [ol-ext@4.0.7](https://github.com/Viglino/ol-ext) - BSD-3-Clause
- [ol-mapbox-style@9.7.0](https://github.com/openlayers/ol-mapbox-style) - BSD-2-Clause
- [ol@7.3.0](https://github.com/openlayers/openlayers) - BSD-2-Clause
- [pako@2.1.0](https://github.com/nodeca/pako) - (MIT AND Zlib)
- [parse-headers@2.0.5](https://github.com/kesla/parse-headers) - MIT
- [parse5@5.1.1](https://github.com/inikulin/parse5) - MIT
- [pbf@3.2.1](https://github.com/mapbox/pbf) - BSD-3-Clause
- [protocol-buffers-schema@3.6.0](https://github.com/mafintosh/protocol-buffers-schema) - MIT
- [quick-lru@6.1.1](https://github.com/sindresorhus/quick-lru) - MIT
- [quickselect@2.0.0](undefined) - ISC
- [rbush@3.0.1](https://github.com/mourner/rbush) - MIT
- [resolve-protobuf-schema@2.1.0](https://github.com/mafintosh/resolve-protobuf-schema) - MIT
- [rw@1.3.3](https://github.com/mbostock/rw) - BSD-3-Clause
- [rxjs@7.8.0](https://github.com/reactivex/rxjs) - Apache-2.0
- [sort-asc@0.1.0](https://github.com/jonschlinkert/sort-asc) - MIT
- [sort-desc@0.1.1](https://github.com/jonschlinkert/sort-desc) - MIT
- [sort-object@0.3.2](https://github.com/helpers/sort-object) - MIT
- [test@1.0.0](undefined) - ISC
- [tiny-emitter@2.1.0](https://github.com/scottcorgan/tiny-emitter) - MIT
- [tslib@2.5.0](https://github.com/Microsoft/tslib) - 0BSD
- [web-worker@1.2.0](https://github.com/developit/web-worker) - Apache-2.0
- [xml-utils@1.7.0](https://github.com/DanielJDufour/xml-utils) - CC0-1.0
- [zone.js@0.12.0](https://github.com/angular/angular) - MIT
```

This is interesting and it might be an issue relatied to Windows. I'll investigate further.

**Comment by RSeidelsohn** [comment]
Hello @eamodio,

do you get your broken results under Windows?

**Comment by RSeidelsohn** [comment]
Hello again, @eamodio and @jakwarrior,

I have a suspect for this issue: From `license-checker-rseidelsohn v4.2.0` to `license-checker-rseidelsohn v4.2.1` I updated the version of the npm module `read-installed-packages` from `1.0` to `2.0`, and this module is directly involved in collecting all installed modules. So for now I have contacted the maintainer of that repo and asked him whether he was aware of any issues related with Windows 10 and directories starting with a `@`.
We'll see. I might as well bring back version `1.0`, but I have to dig deeper into this first.

Cheers, Roman.

**Comment by Semigradsky** [comment]
Hi! In `read-installed-packages` I switched from deprecated `readdir-scoped-modules` to `@npmcli/fs`.

@eamodio @jakwarrior Could you please check that this function ([fs.readdirScoped](https://github.com/npm/fs#fsreaddirscopedroot---promise)) correctly working for you on Windows?

**Comment by eamodio** [comment]
Woo! Fixed in `4.2.6`. Thank you!

**Comment by RSeidelsohn** [comment]
Thanks for the feedback!
Closing issue finally.
